sstables_loader.cc

```diff
@@ -185,6 +185,12 @@ inet_address_vector_replica_set tablet_sstable_streamer::get_endpoints(const dht
         if (!tinfo) {
             return std::nullopt;
         }
+        // Don't write to pending replica if write selector is set to previous, otherwise
+        // a failed migration will not roll back the writes, and that can lead to resurrection
+        // if tablet is later migrated back into the same node.
+        if (tinfo->writes == locator::write_replica_set_selector::previous) {
+            return std::nullopt;
+        }
```

Comment:
This part also seems wrong:

```
if (std::ranges::none_of(current_targets, is_leaving)) {
    return std::nullopt;
}
```

Even if previous endpoints are not leaving, we should stream to the pending endpoint (RF increase). Otherwise, it will be missing data after transition.

We should stream to erm->get_natural_endpoints() + erm->get_pending_endpoints(), that's the current contract. This takes care of the selector. 

For primary replica mode, we should stream to:
 
```
switch (tinfo->writes) {
  case previous: return primary(previous);
  case both: return primary(previous) + primary(next);
  case next: return primary(next);
}
```

may be worth encapsulating in erm as get_primary_endpoints().

Reply:
"Even if previous endpoints are not leaving, we should stream to the pending endpoint (RF increase). Otherwise, it will be missing data after transition." interesting, thanks.

will proceed with the changes.